packages/next/src/cli/next-info.ts

```diff
@@ -124,4 +169,460 @@ const nextInfo: CliCommand = async (argv) => {
   }
 }
 
+/**
+ * Using system-installed tools per each platform, trying to read shared dependencies of next-swc.
+ * This is mainly for debugging DLOPEN failure.
+ *
+ * We don't / can't install these tools by ourselves, will skip the check if we can't find them.
+ */
+async function runSharedDependencyCheck(
+  tools: Array<{ bin: string; checkArgs: Array<string>; args: Array<string> }>,
+  skipMessage: string
+): Promise<TaskResult> {
+  const currentPlatform = os.platform()
+  const spawn = require('next/dist/compiled/cross-spawn')
+  const { getSupportedArchTriples } = require('../build/swc')
+  const triples = getSupportedArchTriples()[currentPlatform]?.[os.arch()] ?? []
+  // First, check if system have a tool installed. We can't install these by our own.
+
+  const availableTools = []
+  for (const tool of tools) {
+    try {
+      const check = spawn.sync(tool.bin, tool.checkArgs)
+      if (check.status === 0) {
+        availableTools.push(tool)
+      }
+    } catch {
+      // ignore if existence check fails
+    }
+  }
+
+  if (availableTools.length === 0) {
+    return {
+      messages: skipMessage,
+      result: 'skipped',
+    }
+  }
+
+  const outputs: Array<string> = []
+  let result: 'pass' | 'fail' = 'fail'
+
+  for (const triple of triples) {
+    const triplePkgName = `@next/swc-${triple.platformArchABI}`
+    let resolved
+    try {
+      resolved = require.resolve(triplePkgName)
+    } catch (e) {
+      return {
+        messages:
+          'Cannot find next-swc installation, skipping dependencies check',
+        result: 'skipped',
+      }
+    }
+
+    for (const tool of availableTools) {
+      const proc = spawn(tool.bin, [...tool.args, resolved])
+      outputs.push(`Running ${tool.bin} ------------- `)
+      // Captures output, doesn't matter if it fails or not since we'll forward both to output.
+      const procPromise = new Promise((resolve) => {
+        proc.stdout.on('data', function (data: string) {
+          outputs.push(data)
+        })
+        proc.stderr.on('data', function (data: string) {
+          outputs.push(data)
+        })
+        proc.on('close', (c: any) => resolve(c))
+      })
+
+      let code = await procPromise
+      if (code === 0) {
+        result = 'pass'
+      }
+    }
+  }
+
+  return {
+    output: outputs.join('\n'),
+    result,
+  }
+}
+
+/**
+ * Collect additional diagnostics information and print it to stdout or write it into a file.
+ *
+ * @param outFile If specified, the output will be written into this file instead of stdout.
+ */
+async function printDiagsInfo(outFile?: string | undefined) {
```

Comment:
```suggestion
 */
async function printVerbose() {
```

Looks like `outFile` is no longer used.

Also lets call this "verbose" now.